spec/index.emu

```diff
@@ -90,16 +106,25 @@ location: https://rawgit.com/tc39/proposal-realms/master/index.html
       <h1>Properties of the Realm Prototype Object</h1>
 
       <emu-clause id="sec-realm.prototype.eval">
-          <h1>Realm.prototype.evaluate ( _sourceText_ )</h1>
+          <h1>Realm.prototype.evaluate ( _sourceText_, _options_ )</h1>
 
-          Synchronously execute a top-level script. The _sourceText_ is interpreted as a Script and evaluated with this bound to the realm's global object.
+          Synchronously execute a top-level script or module. The _sourceText_ is interpreted as a SourceScript or SourceModule and evaluated with this bound to the realm's global object.
 
           <emu-alg>
           1. Let _O_ be *this* value.
           1. Perform ? RequireInternalSlot(_O_, [[Realm]]).
-          1. If Type(_sourceText_) is not String, throw a *TypeError* exception.
           1. Let _realm_ be _O_.[[Realm]].
-          1. Return ? PerformRealmEvaluation(_sourceText_, _realm_).
+          1. Let _sourceTextString_ be ? ToString(_sourceText_).
+          1. Let _sourceType_ be *undefined*.
+          1. If _options_ is not *undefined*, then
+            1. Let _options_ be ? ToObject(_options_).
+            1. Let _sourceType_ be ? Get(_options_, *"type"*).
+          1. If _sourceType_ is *"module"*, then
+            1. Return ? PerformRealmModuleEvaluation(_sourceText_, _realm_).
+          1. Else if _sourceType_ is *"script"*, then
+            1. Return ? PerformRealmScriptEvaluation(_sourceText_, _realm_).
+          1. Else
+            1. Throw a RangeError exception.
```

Comment:
@erights @ljharb for now I'm assuming that we want to throw if `options.type` is not provided or is invalid value. We can definitely relax this, and add a default value. Any thoughts?

```js
const r = new Realm();
r.evaluate(`1`, { type: 'module' });
r.evaluate(`2`, { type: 'script' });
```

Reply:
seems better to keep it explicit, but at that point why have it in the options bag instead of as a required second argument?

Reply:
good question! two main reasons that we have discussed: 

* `evaluate()` is also a thing in evaluator/compartment, just with more stuff on it, like endowments.
* maybe in the future we will have more options here, like strict or sloppy, etc.

But I'm fine either way. We can overload the arguments if we need more options in the future. @erights any opinion here?

Reply:
I'd generally expect options bags to only hold optional arguments; in the case of "evaluate", the parse goal it's evaluating in seems pretty first-class to me. In other words, I'd expect an optional options bag (now or in the future) to be the *third* argument.

Reply:
feedback from the SES meeting yesterday about this question:

* majority of the group will like to have a default value. I still think being explicit about this is a safer (it is more restrictive, and can be relaxed in the future).
* default value to `strict-script`, which is kinda a new concept, their argument was to not favor sloppy mode, which is still possible, but only via `eval` from `globalThis`. No capability lost there, just less convience.
* if there is a default value, then it is better to have the options bag. which I believe all agree, but first lets get to some consensus about that first question.
* there was some discussion about tc53 use cases, and whether or not some folks will simply be able to throw when `type=module` is provided, we need more info about this use-case, and how that impact the API.

Reply:
> @erights did I miss anything?

That's what I remember, yes. For reference, the mtg is at https://www.youtube.com/watch?v=hqKGKDw-FCY&list=PLzDw4TTug5O1jzKodRDp3qec8zl88oxGd

Reply:
Did the SES meeting consider
- Using separate method names, as these really have different types, and for the Promise vs not issue @zenparsing identified?
- Having an expression form, as @erights initially proposed at the start of the thread?